gateway-ha/src/main/java/io/trino/gateway/ha/config/HaGatewayConfiguration.java

```diff
@@ -289,6 +298,19 @@ private void validateStatementPath(String statementPath, List<String> statementP
         }
     }
 
+    public boolean isPathWhiteListed(String path)
+    {
+        List<Pattern> extraWhitelistPaths = requireNonNull(this.getExtraWhitelistPaths()).stream().map(Pattern::compile).collect(toImmutableList());
+        return statementPaths.stream().anyMatch(path::startsWith)
+                || path.startsWith(V1_QUERY_PATH)
+                || path.startsWith(TRINO_UI_PATH)
+                || path.startsWith(V1_INFO_PATH)
+                || path.startsWith(V1_NODE_PATH)
+                || path.startsWith(UI_API_STATS_PATH)
+                || path.startsWith(OAUTH_PATH)
+                || extraWhitelistPaths.stream().anyMatch(pattern -> pattern.matcher(path).matches());
```

Comment:
This code is duplicate at [RoutingTargetHandler](https://github.com/trinodb/trino-gateway/blob/main/gateway-ha/src/main/java/io/trino/gateway/ha/handler/RoutingTargetHandler.java#L121-L131).

If this is needed at early stage, why don't we remove it at RoutingTargetHandler and use this instead?
Also, we could update [RouterPreMatchContainerRequestFilter.java](https://github.com/trinodb/trino-gateway/blob/main/gateway-ha/src/main/java/io/trino/gateway/proxyserver/RouterPreMatchContainerRequestFilter.java)